types.go

```diff
@@ -284,28 +284,115 @@ type ProposalMetaData struct {
 }
 
 type ProposalCensorshipRecord struct {
-	Token     string `json:"token"`
+	Token     string `json:"token" storm:"index"`
 	Merkle    string `json:"merkle"`
 	Signature string `json:"signature"`
 }
 
 type Proposal struct {
+	ID               int                      `storm:"id,increment"`
+	Token            string                   `json:"-" storm:"index"`
+	Category         int32                    `json:"category" storm:"index"`
 	Name             string                   `json:"name"`
-	State            int                      `json:"state"`
-	Status           int                      `json:"status"`
+	State            int32                    `json:"state"`
+	Status           int32                    `json:"status"`
 	Timestamp        int64                    `json:"timestamp"`
 	UserID           string                   `json:"userid"`
 	Username         string                   `json:"username"`
 	PublicKey        string                   `json:"publickey"`
 	Signature        string                   `json:"signature"`
-	NumComments      int                      `json:"numcomments"`
+	NumComments      int32                    `json:"numcomments"`
 	Version          string                   `json:"version"`
 	PublishedAt      int64                    `json:"publishedat"`
 	Files            []ProposalFile           `json:"files"`
 	MetaData         []ProposalMetaData       `json:"metadata"`
 	CensorshipRecord ProposalCensorshipRecord `json:"censorshiprecord"`
+	VoteStatus       ProposalVoteStatus       `json:"votestatus"`
+	VoteSummary      ProposalVoteSummary      `json:"votesummary"`
 }
 
 type Proposals struct {
 	Proposals []Proposal `json:"proposals"`
 }
+
+type ProposalResults struct {
+	Proposal Proposal `json:"proposal"`
+}
+
+type ProposalVoteOption struct {
+	ID          string `json:"id"`
+	Description string `json:"description"`
+	Bits        int32  `json:"bits"`
+}
+
+type ProposalVoteOptionResult struct {
+	Option        ProposalVoteOption `json:"option"`
+	VotesReceived int64              `json:"votesreceived"`
+}
+
+type ProposalVoteStatus struct {
+	Token              string                     `json:"token"`
+	Status             int32                      `json:"status"`
+	TotalVotes         int32                      `json:"totalvotes"`
+	OptionsResult      []ProposalVoteOptionResult `json:"optionsresult"`
+	EndHeight          string                     `json:"endheight"`
+	BestBlock          string                     `json:"bestblock"`
+	NumOfEligibleVotes int32                      `json:"numofeligiblevotes"`
+	QuorumPercentage   int32                      `json:"quorumpercentage"`
+	PassPercentage     int32                      `json:"passpercentage"`
+}
+
+type ProposalVotesStatus struct {
+	VotesStatus []ProposalVoteStatus `json:"votesstatus"`
+}
```

Comment:
I don't think these `ProposalVote-` types are unnecessary, It should be reduced to the basic data needed by the frontend ie
```
type ProposalVoteOption struct {
	YesCount int32
	NoCount  int32
}
```